src/provider/provider_os_memory.c

```diff
@@ -331,6 +331,13 @@ static umf_result_t os_initialize(void *params, void **provider) {
         goto err_destroy_hwloc_topology;
     }
 
+    os_provider->fd_offset_map = critnib_new();
+    if (!os_provider->fd_offset_map) {
+        LOG_ERR("creating file descriptor offset map failed");
+        ret = UMF_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+        goto err_destroy_hwloc_topology;
+    }
+
```

Comment:
what will happen now, the other way around, if e.g. `translate_params` fail. Shouldn't we call `critnib_delete`...?